src/openms/source/ANALYSIS/ID/FragmentIndex.cpp

```diff
@@ -119,68 +116,92 @@ namespace OpenMS
                 continue;
```

Comment:
This check could be done in ModPepGen at some point. Especially if only positive mods are considered you can stop quite early or discard the peptide from the beginning.

Reply:
Also if you merge https://github.com/OpenMS/OpenMS/pull/6859 you will not need to recalculate redundant masses for peptides with the same amount of modifications again.

Mass thresholds could be included in that PR.